File: app/src/main/java/com/ivanovsky/passnotes/util/LocaleUtils.java
package com.ivanovsky.passnotes.util;

import android.content.Context;
import android.os.Build;
import androidx.annotation.NonNull;

import java.util.Locale;

public class LocaleUtils {

	@NonNull
	public static Locale getSystemLocale(@NonNull Context context) {
		Locale result;

		if (Build.VERSION.SDK_INT >= 24) {
			result = context.getResources().getConfiguration().getLocales().get(0);
		} else {
			result = context.getResources().getConfiguration().locale;
		}

		return result;
	}
}

File: app/src/main/kotlin/com/ivanovsky/passnotes/util/LocaleUtils.kt
package com.ivanovsky.passnotes.util

import android.content.Context
import android.os.Build
import java.util.Locale

object LocaleUtils {
    @JvmStatic
    @Suppress("DEPRECATION")
    fun getSystemLocale(context: Context): Locale {
        val configuration = context.resources.configuration
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            configuration.locales[0]
        } else {
            configuration.locale
        }
    }
}